bndtools.core/src/bndtools/preferences/WorkspaceOfflineChangeAdapter.java

```diff
@@ -0,0 +1,19 @@
+package bndtools.preferences;
+
+import org.eclipse.jface.util.IPropertyChangeListener;
+import org.eclipse.jface.util.PropertyChangeEvent;
+
+public abstract class WorkspaceOfflineChangeAdapter implements IPropertyChangeListener, WorkspaceOfflineChangeListener {
+    @Override
+    public final void propertyChange(PropertyChangeEvent event) {
+        if (!event.getProperty().equals(BndPreferences.PREF_WORKSPACE_OFFLINE)) {
+            return;
+        }
+        workspaceOfflineChanged((boolean) event.getNewValue());
```

Comment:
@rotty3000 points out you have an anonymous subclass below which override the following method. Perhaps then, this type should not implement the methods. It is already an abstract class.